scripts/o3de/tests/test_manifest.py

```diff
@@ -147,3 +144,9 @@ def validate_gem_json(gem_json_path) -> bool:
 
             # make sure the o3de manifest isn't attempted to be loaded
             load_o3de_manifest_patch.assert_not_called()
+
+class TestGetAllGems:
+    def test_get_all_gems_returns_unique_paths(self):
+        all_gems = manifest.get_all_gems()
+        unique_gems =  list(dict.fromkeys(all_gems))
+        assert len(all_gems) == len(unique_gems)
```

Comment:
This test is actually reading from the manifest files on disk and not mocking the calls.
This opens up a point of failure in AR, with being unable to open the o3de_manifest.json file for some reason.

This test needs to be part ofthe framework above to make sure that it doesn't rely on reading non-temp files on disk

Reply:
Yup, I was just using this for local testing and didn't mean to commit it - I'll turn it into some proper mocked tests.